tests/api_test.py

```diff
@@ -1828,6 +1831,7 @@ def f(z):
     expected = grad(f)(zs)
     self.assertAllClose(ans, expected)
 
+  @jax.numpy_dtype_promotion('standard')  # Test explicitly exercises implicit dtype promotion.
```

Comment:
Here, the test is explicitly testing heterogenous inputs (it's even in the test name!) so changing to homogeneous inputs seems like the wrong decision.